### PR TITLE
Remove login from root page

### PR DESCRIPTION
**Pull request by spaniernathan**
Removing the login implies that:
- Guest users can fetch data from the api
- Video no longer need token to be player

### CODE REVIEW

File: amplify/backend/storage/amplifyVideoCommuAssets/s3-cloudformation-template.json [hunk 1]
```diff
@@ -162,7 +162,7 @@
 	"Resources": {
 		"S3Bucket": {
 			"Type": "AWS::S3::Bucket",
-
+
```
Comment (florian-vuillemot) [inline amplify/backend/storage/amplifyVideoCommuAssets/s3-cloudformation-template.json:165]: Useless spaces in all this file

File: src/shared/components/VideoPlayer.tsx [hunk 2]
```diff
@@ -28,7 +24,7 @@ export default class VideoPlayer extends React.Component<VideoPlayerPropsInferfa
     componentDidMount() {
         // eslint-disable-next-line
         videojs.Vhs.xhr.beforeRequest = (options: any) => {
-            options.uri = `${options.uri}${this.options?.token}`
+            options.uri = `${options.uri}`
```
Comment (florian-vuillemot) [inline src/shared/components/VideoPlayer.tsx:27]: Useless assignment ?

File: src/shared/utilities/fetch.ts [hunk 3]
```diff
@@ -1,25 +1,34 @@
...
 
 async function fetchSection(id: string | null) {
-    return API.graphql(
-        graphqlOperation(getSection, { input: { id } })
-    ) as GraphQLResult<APIt.GetSectionQuery>
+    return API.graphql({
+        query: getSection,
+        variables: { input: { id } },
+        authMode:
+            Auth.Credentials.getCredSource() !== 'userPool'
```
Comment (florian-vuillemot) [inline src/shared/utilities/fetch.ts:28]: Duplicate code with fetchSections?
```
Auth.Credentials.getCredSource() !== 'userPool'
     ? GRAPHQL_AUTH_MODE.AWS_IAM
     : GRAPHQL_AUTH_MODE.AMAZON_COGNITO_USER_POOLS
```

File: src/shared/utilities/vod-fetch.ts [hunk 4]
```diff
@@ -44,13 +43,24 @@ const listVodAssets = /* GraphQL */ `
...
+                    : GRAPHQL_AUTH_MODE.AMAZON_COGNITO_USER_POOLS,
+        }) as GraphQLResult<APIt.ListVodAssetsQuery>
     else
-        return API.graphql(
-            graphqlOperation(listVodAssets)
-        ) as GraphQLResult<APIt.ListVodAssetsQuery>
+        return API.graphql({
+            query: listVodAssets,
+            authMode:
+                Auth.Credentials.getCredSource() !== 'userPool'
```
Comment (florian-vuillemot) [inline src/shared/utilities/vod-fetch.ts:60]: Duplicate code in the same file + fetch.ts

File: src/shared/utilities/vod-mutate.ts [hunk 5]
```diff
@@ -27,6 +27,61 @@ async function createVideoAsset(payload: APIt.CreateVodAssetInput) {
...
+    })
+}
+
+async function putThumbnailFile(
+    file: File,
+    id: string,
+    thumbnailExtension: string[]
+) {
+    return Storage.put(
+        `thumbnails/${id}.${thumbnailExtension[thumbnailExtension.length - 1]}`,
```
Comment (florian-vuillemot) [inline src/shared/utilities/vod-mutate.ts:49]: it can be undefined, is that ok?

File: src/shared/utilities/vod-mutate.ts [hunk 6]
```diff
@@ -27,6 +27,61 @@ async function createVideoAsset(payload: APIt.CreateVodAssetInput) {
...
+        }
+    )
+}
+
+async function setThumbnailObject(id: string, thumbnailExtension: string[]) {
+    return API.graphql(
+        graphqlOperation(createThumbnailObject, {
+            input: {
+                id: id,
+                ext: thumbnailExtension[thumbnailExtension.length - 1],
```
Comment (florian-vuillemot) [inline src/shared/utilities/vod-mutate.ts:69]: it can be undefined, is that ok?

File: src/shared/utilities/vod-mutate.ts [hunk 7]
```diff
@@ -39,78 +94,33 @@ const uploadVideo = async (
...
-                    )
-                },
-            }
-        )
+        await putVodFile(vodFile, id, vodExtension)
     } catch (error) {
-        console.error('vod-mutate(storage put vod file): ', error)
-        return
+        console.error('vod-mutate.ts(putVodFile): ', error)
+        // return
```
Comment (florian-vuillemot) [inline src/shared/utilities/vod-mutate.ts:100]: useless ?

File: src/shared/utilities/vod-mutate.ts [hunk 8]
```diff
@@ -39,78 +94,33 @@ const uploadVideo = async (
...
-                    )
-                },
-            }
-        )
+        await putThumbnailFile(thumbnailFile, id, thumbnailExtension)
     } catch (error) {
-        console.error('vod-mutate(storage put thumbnail file): ', error)
-        return
+        console.error('vod-mutate.ts(putThumbnailFile): ', error)
+        // return
```
Comment (florian-vuillemot) [inline src/shared/utilities/vod-mutate.ts:107]: useless?

**Review by florian-vuillemot — CHANGES_REQUESTED** [review]

File: src/shared/components/VideoPlayer.tsx (hunk 2 above)
Comment (jgchoppe) [reply src/shared/components/VideoPlayer.tsx:27]: The video isn't working without this assignment

File: src/shared/utilities/vod-mutate.ts [hunk 9]
```diff
@@ -39,78 +94,33 @@ const uploadVideo = async (
...
-                    ext: thumbnailExtension[thumbnailExtension.length - 1],
-                },
-            })
-        )
+        await setThumbnailObject(id, thumbnailExtension)
     } catch (error) {
-        console.error('vod-mutate(createThumbnail): ', error)
-        return
+        console.error('vod-mutate.tx(setThumbnailObject): ', error)
+        // return
```
Comment (jgchoppe) [inline src/shared/utilities/vod-mutate.ts:114]: useless?

File: src/shared/utilities/vod-mutate.ts [hunk 10]
```diff
@@ -39,78 +94,33 @@ const uploadVideo = async (
...
-                    id: id,
-                },
-            })
-        )
+        await setVideoObject(id)
     } catch (error) {
-        console.error('vod-mutate(createVideo): ', error)
-        return
+        console.error('vod-mutate.tx(setVideoObject): ', error)
+        // return
```
Comment (jgchoppe) [inline src/shared/utilities/vod-mutate.ts:120]: useless?

**Review by jgchoppe — CHANGES_REQUESTED** [review]
Should we also remove all console log ? @florian-vuillemot @spaniernathan

File: amplify/backend/storage/amplifyVideoCommuAssets/s3-cloudformation-template.json (hunk 1 above)
Comment (spaniernathan) [reply amplify/backend/storage/amplifyVideoCommuAssets/s3-cloudformation-template.json:165]: those are auto-generated files

**Pull request merged**